File: backend/app/models/todo_attachment.py
from datetime import datetime

from sqlalchemy import ForeignKey, String
from sqlalchemy.orm import Mapped, mapped_column

from ..database import Base, utcnow


class TodoAttachment(Base):
    __tablename__ = "todo_attachments"

    id: Mapped[int] = mapped_column(primary_key=True)
    todo_id: Mapped[int] = mapped_column(
        ForeignKey("todos.id", ondelete="CASCADE"), index=True
    )
    filename: Mapped[str] = mapped_column(String(255))
    stored_path: Mapped[str] = mapped_column(String(500))
    content_type: Mapped[str] = mapped_column(String(100), default="application/octet-stream")
    file_size: Mapped[int] = mapped_column(default=0)
    created_at: Mapped[datetime] = mapped_column(default=utcnow)

File: flutter/lib/features/todos/domain/todo_attachment.dart
class TodoAttachment {
  final int id;
  final String filename;
  final String contentType;
  final int fileSize;
  final DateTime createdAt;
  final String? downloadUrl;

  const TodoAttachment({
    required this.id,
    required this.filename,
    required this.contentType,
    required this.fileSize,
    required this.createdAt,
    this.downloadUrl,
  });

  factory TodoAttachment.fromJson(Map<String, dynamic> json) {
    return TodoAttachment(
      id: json['id'] as int,
      filename: json['filename'] as String,
      contentType:
          json['content_type'] as String? ?? 'application/octet-stream',
      fileSize: json['file_size'] as int? ?? 0,
      createdAt: DateTime.parse(json['created_at'] as String),
      downloadUrl: json['download_url'] as String?,
    );
  }
}
